src/main/java/com/danielagapov/spawn/Models/User.java

```diff
@@ -18,12 +20,13 @@ public class User implements Serializable {
         @Id
         @GeneratedValue
         private UUID id;
-
+        private List<User> friends;
```

Comment:
Leave this out of the entity. It should only be in the DTO. Refer to the README's DTO drawing, as this case happens to exactly match it. 

Basically, `friends` as a list can't actually be put directly into SQL, which is what this `@Entity` tag does behind the hood with Spring Data JPA to a MySQL table. 

So, while we need the `friends` property in the DTO, we need to fetch that appropriately in the mapper files, not from the entity directly.

Reply:
As a rule-of-thumb, you should never have `List`s in `@Entity` objects, as those can't be translated into SQL

Reply:
Upon further research, I think that the easiest solution is to use a series of `@JoinColumn` and `@ManyToOne` (and its variants) annotations. This will join the tables allowing us to directly have collections in our entities. The problem is that I have no solid grasp of how these join operations work and we have no testing infrastructure in place to ensure that these annotations will behave as we expect granted that we annotate our fields correctly.

Reply:
I disagree with this approach. I think we shouldn't have to touch the entities at all, actually, and that'd be easiest. Instead, we can aggregate related entities in the service classes, to then put into the DTO, with arguments. Please see my related comments in your mapping files, like in `FriendTagMapper.java`, for example.

Reply:
Sounds good, I was messing around with this all of yesterday and I couldn't get it to work because of the cyclical dependencies when converting to DTO or to entity. It turns out a lot of this commit you've reviewed doesn't even work when I was testing :/